File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        map<int,int>win,lose;
        int i,n=matches.size();
        for(i=0;i<n;i++)
        {
            win[matches[i][0]]++;
            lose[matches[i][1]]++;
        }
        vector<vector<int>>ans(2);
        for(auto u:win)
        {
            if(lose[u.first]>0)
               continue;
            else ans[0].push_back(u.first);
        }
        for(auto u:lose)
        {
            if(lose[u.first]==1)
                ans[1].push_back(u.first);
        }
        return ans;
        
    }
};

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.js
/**
 * @param {number[][]} matches
 * @return {number[][]}
 */
var findWinners = function(matches) {
    var hashMap ={};
    var ans = [[],[]];
    for(mat of matches)
    {
        if(hashMap[mat[1]])
            hashMap[mat[1]]++;
        else hashMap[mat[1]]=1;
        if(hashMap[mat[0]])
            hashMap[mat[0]]+=0;
        else hashMap[mat[0]]=0;
    }
    //console.log(hashMap);
    for(var item in hashMap)
    {
        if(hashMap[item]==0)
            ans[0].push(item);
        else if(hashMap[item]==1)
            ans[1].push(item);
    }
    return ans;
};